ra/ra.go

```diff
@@ -751,28 +749,100 @@ func domainsForRateLimiting(names []string) ([]string, error) {
 	return domains, nil
 }
 
+// suffixesForRateLimiting transforms a list of FQDNs into a list of exact
+// public suffix matches for the purpose of rate limiting. It also
+// de-duplicates the output domains. Domains that are not an exact match for a
```

Comment:
This comment can be rewritten to be clearer. E.g. "... returns the subset of input names that are exactly equal to a public suffix."